app/services/remote_tagging_service.rb

```diff
@@ -0,0 +1,70 @@
+require 'faraday'
+class RemoteTaggingService
+  def initialize(options)
+    @app_name = options[:app_name]
+    @object_type = options[:object_type]
+    @object_id = options[:object_id]
+  end
+
+  def process(action, tag)
+    post_request(action_url(action), tag)
+    self
+  end
+
+
+  def self.remotes
+    [{ :app_name => 'topology', :object_type => 'ServiceInventory', :url => proc {topo_url} },
+     { :app_name => 'topology', :object_type => 'Credential',       :url => proc {topo_url} },
+     { :app_name => 'catalog',  :object_type => 'PortfolioItem',    :url => proc {catalog_url} },
+     { :app_name => 'catalog',  :object_type => 'Portfolio',        :url => proc {catalog_url} },
+     { :app_name => 'sources',  :object_type => 'Source',           :url => proc {sources_url} }]
```

Comment:
Let's consider using OpenShift ConfigMap to store the mapping from app/type -> url.
The key can even include the action, such as `add/topology/ServiceInventory`.

The url can contain an object_id placeholder. The in our code we only need to do a replacement to get a fully qualified URL.

Reply:
@bzwei Future